controllers/admin/AdminProductsController.php

```diff
@@ -2446,7 +2446,21 @@ public function processSuppliers($id_product = null)
             foreach ($associated_suppliers as $key => $associated_supplier) {
                 /** @var ProductSupplier $associated_supplier */
                 if (!in_array($associated_supplier->id_supplier, $suppliers_to_associate)) {
-                    $associated_supplier->delete();
+                    // This specific delete is made to ensure product v1/v2 consistency for product / supplier association
+                    // Cf. https://github.com/PrestaShop/PrestaShop/pull/26609
+
+                    $res = $associated_supplier->delete();
+
```

Comment:
Add a comment like "This code block was moved from ProductSupplier::delete() to avoid side effects in product page v2 while keeping same behavior in product page v1"